src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -331,32 +338,41 @@ JNI_METHOD(void, beginSendCommand)(JNIEnv * env, jobject self, jlong deviceContr
     static const size_t bufferSize = 1024;
     auto * buffer                  = System::PacketBuffer::NewWithAvailableSize(bufferSize);
 
-    // Hardcode endpoint to 1 for now
-    uint8_t endpoint = 1;
-
-    uint16_t dataLength = 0;
-    switch (commandID)
+    if (buffer == nullptr)
     {
-    case 0:
-        dataLength = encodeOffCommand(buffer->Start(), bufferSize, endpoint);
-        break;
-    case 1:
-        dataLength = encodeOnCommand(buffer->Start(), bufferSize, endpoint);
-        break;
-    case 2:
-        dataLength = encodeToggleCommand(buffer->Start(), bufferSize, endpoint);
-        break;
-    default:
-        ChipLogError(Controller, "Unknown command: %d", commandID);
-        return;
+        err = CHIP_ERROR_NO_MEMORY;
+        pthread_mutex_unlock(&sStackLock);
     }
+    else
+    {
+
+        // Hardcode endpoint to 1 for now
+        uint8_t endpoint = 1;
+
+        uint16_t dataLength = 0;
+        switch (commandID)
+        {
+        case 0:
+            dataLength = encodeOffCommand(buffer->Start(), bufferSize, endpoint);
+            break;
+        case 1:
+            dataLength = encodeOnCommand(buffer->Start(), bufferSize, endpoint);
+            break;
+        case 2:
+            dataLength = encodeToggleCommand(buffer->Start(), bufferSize, endpoint);
+            break;
+        default:
+            ChipLogError(Controller, "Unknown command: %d", commandID);
+            return;
+        }
 
-    buffer->SetDataLength(dataLength);
+        buffer->SetDataLength(dataLength);
 
-    // Hardcode endpoint to 1 for now
-    err = deviceController->SendMessage((void *) "SendMessage", buffer);
+        // Hardcode endpoint to 1 for now
+        err = deviceController->SendMessage((void *) "SendMessage", buffer);
 
-    pthread_mutex_unlock(&sStackLock);
+        pthread_mutex_unlock(&sStackLock);
```

Comment:
nit: factor this out

Reply:
I created a 'scoped lock' class to be more C++-like in this file.